p2panda-rs/src/schema/schema.rs

```diff
@@ -0,0 +1,189 @@
+            .try_into()
+            .unwrap();
+
+        // Create first schema field "is_accessible"
+        // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+        let mut bool_field = BTreeMap::new();
+        bool_field.insert(
+            "name".to_string(),
+            OperationValue::Text("is_accessible".to_string()),
+        );
+        bool_field.insert("type".to_string(), OperationValue::Text("bool".to_string()));
+
+        let bool_field_view_id = DocumentViewId::new(vec![relation_operation_id_1]);
+        let bool_field_view: SchemaFieldView =
+            DocumentView::new(bool_field_view_id, bool_field_id, bool_field)
+                .try_into()
+                .unwrap();
+
+        // Create second schema field "capacity"
+        // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+        let mut capacity_field = BTreeMap::new();
+        capacity_field.insert(
+            "name".to_string(),
+            OperationValue::Text("capacity".to_string()),
+        );
+        capacity_field.insert("type".to_string(), OperationValue::Text("int".to_string()));
+
+        let capacity_field_view_id =
+            DocumentViewId::new(vec![relation_operation_id_2, relation_operation_id_3]);
+        let capacity_field_view: SchemaFieldView =
+            DocumentView::new(capacity_field_view_id, capacity_field_id, capacity_field)
+                .try_into()
+                .unwrap();
+
+        // Create venue schema from schema and field views
+        // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+        assert!(Schema::new(
+            schema_view.clone(),
+            vec![bool_field_view.clone(), capacity_field_view.clone()]
+        )
+        .is_ok());
+
+        // Invalid fields should fail
+        // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+        let mut toilet_size_field = BTreeMap::new();
+        toilet_size_field.insert(
+            "name".to_string(),
+            OperationValue::Text("toilet_size".to_string()),
+        );
+        toilet_size_field.insert("type".to_string(), OperationValue::Text("int".to_string()));
+
+        let toilet_size_field_view_id = DocumentViewId::new(vec![relation_operation_id_4]);
+        let toilet_size_field_view: SchemaFieldView = DocumentView::new(
+            toilet_size_field_view_id,
+            toilet_size_field_id,
+            toilet_size_field,
+        )
+        .try_into()
+        .unwrap();
+
+        // Invalid field passed
+        assert!(Schema::new(
+            schema_view.clone(),
+            vec![bool_field_view.clone(), toilet_size_field_view.clone()]
+        )
+        .is_err());
+        // Too few fields passed
+        assert!(Schema::new(schema_view.clone(), vec![bool_field_view.clone()]).is_err());
+        // Too many fields passed
+        assert!(Schema::new(
+            schema_view,
+            vec![bool_field_view, capacity_field_view, toilet_size_field_view]
+        )
+        .is_err());
```

Comment:
Noticed this in other parts of our code as well: We have a tendency to have very long singular tests doing many different things. Sometimes I have a hard time understanding what a test actually does. I'd be in favour of separating them into smaller tests, giving them names which describe the one thing they want to test (or add comments on top), making sure we frame every test around one thing as good as possible.

I know that usually putting all into one test comes from the work one needs to put in to bootstrap the test, but I think this can be made easier with a) using fixtures b) adding more fixtures c) writing test helpers on top of all tests to have reusable methods.